src/main/java/com/google/sps/servlets/NotificationServlet.java

```diff
@@ -0,0 +1,209 @@
+    // user ID.
+    String query =  "SELECT message, url, date_time FROM Notification WHERE id IN " +
+                    "(SELECT notification_id FROM UserNotification WHERE user_id = " + userId +
+                    ") ORDER BY date_time DESC";
+    List<Notification> notifications = new ArrayList<>();
+    // Query the information from tables and create notification object to be stored in ArrayList.
+    try (Connection connection = DriverManager.getConnection(Utility.SQL_LOCAL_URL,
+                                                             Utility.SQL_LOCAL_USER,
+                                                             Utility.SQL_LOCAL_PASSWORD);
+        PreparedStatement pst = connection.prepareStatement(query);
+        ResultSet rs = pst.executeQuery()) {
+      // Iterate through the result of the query to populate the ArrayList and return it as JSON.
+      while(rs.next()){
+        Notification notification = new Notification();
+        notification.message = rs.getString(1);
+        notification.url = rs.getString(2);
+        notification.timestamp = rs.getTimestamp(3);
+        // Stores object in ArrayList.
+        notifications.add(notification);
+      }
+      connection.close();
+    } catch (SQLException ex) {
+        Logger lgr = Logger.getLogger(DataServlet.class.getName());
+        lgr.log(Level.SEVERE, ex.getMessage(), ex);
+    }
+    return notifications;
+  }
+
+  /**
+  * Function that receives the data for a notification and inserts it into the Notification table.
+  */
+  private void insertToNotification(Connection connection, String message, String elementUrl,
+        Timestamp dateTime) {
+    String query = "INSERT INTO Notification(message, url, date_time) VALUES(?,?,?)";
+    try {
+      // Prepare the statement to be inserted.
+      PreparedStatement prepStatement = connection.prepareStatement(query);
+      prepStatement.setString(1, message);
+      prepStatement.setString(2, elementUrl);
+      prepStatement.setTimestamp(3, dateTime);
+      prepStatement.executeUpdate();
+    } catch (SQLException e) {
+      System.out.println(e.getMessage());
+    }
+  }
+
+  /**
+  * Function that receives the data for a notification-user relation and inserts it into
+  * UserNotification.
+  */
+  private void insertToUserNotification(Connection connection, int userId, int notificationId) {
+    String query = "INSERT INTO UserNotification(user_id, notification_id) VALUES(?,?)";
+    try {
+      // Prepare the statement to be inserted.
+      PreparedStatement prepStatement = connection.prepareStatement(query);
+      prepStatement.setInt(1, userId);
+      prepStatement.setInt(2, notificationId);
+      prepStatement.executeUpdate();
+    } catch (SQLException e) {
+      System.out.println(e.getMessage());
+    }
+  }
+
+  /**
+  * Return the id of a just inserted notification.
+  */
+  private int getNotificationId(Connection connection, Timestamp date) {
```

Comment:
Delete `connection` variable if you are not using it inside the function.

Reply:
It is being used.

Reply:
👍